enterprise/replicasync/replicasync.go

```diff
@@ -61,6 +61,7 @@ func New(ctx context.Context, logger slog.Logger, db database.Store, pubsub data
 	if err != nil {
 		return nil, xerrors.Errorf("ping database: %w", err)
 	}
+	// nolint:gocritic // Inserting a replica is a system function.
```

Comment:
Since this function is expecting a system context, perhaps it should be renamed as such ("dangerousSystemCtx", sam as in the other place)? Just wondering if there's a chance this ctx can be used for actions that shouldn't be elevated. Perhaps the manager is extended with new functionality and invoked via HTTP handler ctx and should use those perms?

Reply:
Good point. Modified to set the elevated ctx in the package directly instead of passing in an elevated ctx.